### PR TITLE
 Add tests for tradingPairCutoffs

### PR DESCRIPTION
**Pull request by xiaoruby**

### CODE REVIEW

File: test/testLoopringProtocolImpl.ts [hunk 1]
```diff
@@ -831,4 +865,15 @@ contract("LoopringProtocolImpl", (accounts: string[]) => {
...
   });
 
+  describe("setTradingPairCutoff", () => {
+    it("should be able to set trading pair cutoff timestamp for msg sender", async () => {
+      const ring = await ringFactory.generateSize2Ring01(order1Owner, order2Owner, ringOwner);
+      const order = ring.orders[0];
+      await loopringProtocolImpl.setTradingPairCutoff(new BigNumber(1508566125), order.params.amountS, order.params.amountB, {from: order1Owner});
+      const tradingPairCutoff = await loopringProtocolImpl.tradingPairCutoffs(order.params.amountS, order1Owner);
+
+      // TODO: how to implment the assert when getTradingPairId is an internal function?
```
Comment (xiaoruby) [inline test/testLoopringProtocolImpl.ts]: @dong77 Could you help me with the implementation of the assert? Thank you.

**Comment by xiaoruby** [comment]
#219

File: test/testLoopringProtocolImpl.ts (hunk 1 above)
Comment (dong77) [reply test/testLoopringProtocolImpl.ts]: You need to reimplement the method  using JS. Check out this one: 

https://github.com/jimmychang851129/solidity_test/blob/27891dd41d1bd403eabf795613d3587880fce846/ethereum/contract/setjson.js

```// var keccak256 = require('js-sha3').keccak256;```

Comment (dong77) [reply test/testLoopringProtocolImpl.ts]: please also change "setCutoff" to "cancelOrders" and "setTradingPairCutoff" to "cancelOrders" as well if overloading is permitted.

Comment (xiaoruby) [reply test/testLoopringProtocolImpl.ts]: Truffle does not support function overloading. https://github.com/trufflesuite/truffle/issues/737 So I keep "setCutoff" unchanged and change "setTradingPairCutoff" to "cancelOrders"

File: test/testLoopringProtocolImpl.ts [hunk 2]
```diff
@@ -831,4 +869,22 @@ contract("LoopringProtocolImpl", (accounts: string[]) => {
...
+      await loopringProtocolImpl.cancelOrders(new BigNumber(1508566125), order.params.amountS, order.params.amountB, {from: order1Owner});
+     
+      const token1_256 = keccak256(order.params.amountS.toString(10));;
+      const token2_256 = keccak256(order.params.amountB.toString(10));;
+
+      const tradingPairIdBigInt = bigInt(token1_256, 16).xor(bigInt(token2_256, 16));
+      const tradingPairId = web3.toHex(tradingPairIdBigInt.toString(10));
+      const tradingPairCutoff = await loopringProtocolImpl.tradingPairCutoffs(order1Owner, tradingPairId);
+
+      // FIXME: tradingPairCutoff.toNumber() is 0
```
Comment (xiaoruby) [inline test/testLoopringProtocolImpl.ts]: @dong77 I tried to use keccak256 in the test. Could you help me with the PR? It seems that I didn't implement the method to get trading pair id correctly. Thank you.

File: test/testLoopringProtocolImpl.ts [hunk 3]
```diff
@@ -831,4 +869,20 @@ contract("LoopringProtocolImpl", (accounts: string[]) => {
...
+  describe("cancelOrders", () => {
+    it("should be able to set trading pair cutoff timestamp for msg sender", async () => {
+      const ring = await ringFactory.generateSize2Ring01(order1Owner, order2Owner, ringOwner);
+      const order = ring.orders[0];
+      await loopringProtocolImpl.cancelOrders(order.params.tokenS, order.params.tokenB, new BigNumber(1558566125), {from: order1Owner});
+     
+      const token1_256 = keccak256(order.params.tokenS);
+      const token2_256 = keccak256(order.params.tokenB);
+      const combinedTokenHash = xor(new Buffer(token1_256), new Buffer(token2_256));
+      const tradingPairCutoff = await loopringProtocolImpl.tradingPairCutoffs(order1Owner, [...combinedTokenHash]); //, combinedTokenHash.toString("hex"));
```
Comment (xiaoruby) [inline test/testLoopringProtocolImpl.ts]: Covert a Buffer to bytes array.

**Pull request closed without merging**